### PR TITLE
Enrich CSV reader config: quote & escape

### LINKED ISSUE #6854: Enrich CSV reader config: `quote` & `escape`
### Is your feature request related to a problem or challenge?

Currently CSV reader doesn't allow to set: `quote` & `escape` chars.

### Describe the solution you'd like

`CsvReadOptions` should contains methods to set `quote` & `escape` (like `delimiter`)

For example, how it works in `csv` crate: [escape](https://docs.rs/csv/latest/csv/struct.ReaderBuilder.html#method.escape) & [quote](https://docs.rs/csv/latest/csv/struct.ReaderBuilder.html#method.quote)

### Describe alternatives you've considered

_No response_

### Additional context

Escaping is optional stuff, usually it's just double usage of quote char, but it would be nice to be able to override.

### PR DESCRIPTION
**Pull request by parkma99**
# Which issue does this PR close?

Closes #6854 

# Rationale for this change

# What changes are included in this PR?

Current add quote to CSV reader config.

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by parkma99** [comment]
Hello @alamb , I read the  code in 
https://github.com/apache/arrow-rs/blob/6e0faaf24d84cded196be3abfccd80a154a92740/arrow-csv/src/reader/mod.rs#L1860

in comment says `escape` default is None.

I would like add `escape` as like `delimiter`, got a question how to set default `escape`.

`delimiter` default value is `,` and `quote` default value is `"`. In this line, comment says `escape` default value is None😢.

**Comment by alamb** [comment]
Hi @parkma99  I don't really follow your question

> Set escape as a Option<u8> is a chose. In this way, I perfer to change delimiter and quote to Option<u8> just like

I think this sounds reasonable to me (and if it is `None` fall back to whatever the arrow default is)

**Comment by Weijun-H** [comment]
LGTM 👍

**Review by Weijun-H — APPROVED** [review]

File: datafusion/core/src/datasource/physical_plan/csv.rs [hunk 1]
```diff
@@ -656,7 +682,14 @@ mod tests {
         let mut config = partitioned_csv_config(file_schema, file_groups)?;
         config.projection = Some(vec![0, 2, 4]);
 
-        let csv = CsvExec::new(config, true, b',', file_compression_type.to_owned());
+        let csv = CsvExec::new(
+            config,
+            true,
+            b',',
+            b'"',
+            None,
```
Comment (alamb) [inline datafusion/core/src/datasource/physical_plan/csv.rs:690]: I wonder if we can reduce the number of arguments to `CsvExec::new()` somehow (like maybe pass in the config 🤔 )

**Review by alamb — COMMENTED** [review]
Thanks @parkma99  and @Weijun-H  -- I agree this code looks great.

Can you please add tests (ideally in sqllogictest) showing this new feature working end to end ('CREATE EXTERNAL TABLE ... ` or maybe via the DataFrame interface and then selecting)?

**Review by alamb — APPROVED** [review]
Looks good to me -- thank you @parkma99

**Comment by alamb** [comment]
Since this PR has been open for a while I will merge it. @parkma99  let me know if you want to address @jackwener 's comments in https://github.com/apache/arrow-datafusion/pull/6927#discussion_r1272441251 in this PR otherwise I'll merge it in

File: datafusion/proto/proto/datafusion.proto [hunk 2]
```diff
@@ -88,6 +88,10 @@ message ProjectionColumns {
 message CsvFormat {
   bool has_header = 1;
   string delimiter = 2;
+  string quote = 3;
+  oneof optional_escape {
```
[the remark this answers is not in the document]
Comment (alamb) [reply datafusion/proto/proto/datafusion.proto:92]: I think it is necessary to encode the fact that the escape might be `None` or might be `Some(..)` -- the other fields like `quote` are always some particular value

**Comment by alamb** [comment]
In an attempt to clear the PR backlog, I took the liberty of merging this PR up from main and plan to merge it when CI passes

**Pull request merged**